test/dynamo/test_misc.py

```diff
@@ -5161,6 +5164,13 @@ def fn(x, y):
             res = opt_fn(x, y)
             self.assertTrue(same(ref, x + 1))
             self.assertTrue(same(res, x - 1))
+
+            # we guard on builtins being modified
+            builtins.isinstance = patched_isinstance_2
+            assert id(builtins.isinstance) == id(patched_isinstance_2)
+            assert builtins.isinstance.__code__.co_name == "patched_isinstance_2"
+            res2 = opt_fn(x, y)
+            self.assertTrue(same(res2, x + 1))
```

Comment:
Hmm, could you also test defining a function named isinstance which overrides the builtins by having more precedence in scope?